letsencrypt/client/revoker.py

```diff
@@ -6,146 +14,513 @@
+        self.backup_key_path = backup_key
+
+    def get_installed_msg(self):
+        """Access installed message."""
+        return ", ".join(self.installed)
+
+    def get_subject(self):
+        """Get subject."""
+        return self.cert.get_subject().as_text()
+
+    def get_cn(self):
+        """Get common name."""
+        return self.cert.get_subject().CN
+
+    def get_issuer(self):
+        """Get issuer."""
+        return self.cert.get_issuer().as_text()
+
+    def get_fingerprint(self):
+        """Get sha1 fingerprint."""
+        return self.cert.get_fingerprint(md="sha1")
+
+    def get_not_before(self):
+        """Get not_valid_before field."""
+        return self.cert.get_not_before().get_datetime()
+
+    def get_not_after(self):
+        """Get not_valid_after field."""
+        return self.cert.get_not_after().get_datetime()
+
+    def get_serial(self):
+        """Get serial number."""
+        self.cert.get_serial_number()
+
+    def get_pub_key(self):
+        """Get public key size."""
+        # .. todo:: M2Crypto doesn't support ECC, this will have to be updated
+        return "RSA " + str(self.cert.get_pubkey().size() * 8)
+
+    def get_san(self):
+        """Get subject alternative name if available."""
+        try:
+            return self.cert.get_ext("subjectAltName").get_value()
+        except LookupError:
+            return ""
+
+    def __str__(self):
+        text = []
+        text.append("Subject: %s" % self.get_subject())
+        text.append("SAN: %s" % self.get_san())
+        text.append("Issuer: %s" % self.get_issuer())
+        text.append("Public Key: %s" % self.get_pub_key())
+        text.append("Not Before: %s" % str(self.get_not_before()))
+        text.append("Not After: %s" % str(self.get_not_after()))
+        text.append("Serial Number: %s" % self.get_serial())
+        text.append("SHA1: %s%s" % (self.get_fingerprint(), os.linesep))
+        text.append("Installed: %s" % self.get_installed_msg())
+
+        if self.orig is not None:
+            if self.orig.status == "":
+                text.append("Path: %s" % self.orig.path)
+            else:
+                text.append("Orig Path: %s (%s)" % self.orig)
+        if self.orig_key is not None:
+            if self.orig_key.status == "":
+                text.append("Auth Key Path: %s" % self.orig_key.path)
+            else:
+                text.append("Orig Auth Key Path: %s (%s)" % self.orig_key)
+
+        text.append("")
+        return os.linesep.join(text)
+
+    def pretty_print(self):
+        """Nicely frames a cert str"""
+        frame = "-" * (display_util.WIDTH - 4) + os.linesep
+        return "{frame}{cert}{frame}".format(frame=frame, cert=str(self))
 
-        shutil.copy2(list_file2, list_file)
-        os.remove(list_file2)
-        os.remove(cert["backup_cert_file"])
-        os.remove(cert["backup_key_file"])
+    def __eq__(self, other):
+        return self.cert.as_der() == other.cert.as_der()
```

Comment:
you probably want to check `isinstance(other, Cert)` before trying to access attributes of `other`